packages/dds/tree/src/test/simple-tree/simpleTree.bench.ts

```diff
@@ -0,0 +1,186 @@
+			let tree: DeepTreeNode;
+			benchmark({
+				type: benchmarkType,
+				title: `Update value at leaf of ${numberOfNodes} deep tree`,
+				before: () => {
+					// Setup
+					tree = generateDeepSimpleTree(numberOfNodes, leafValue);
+				},
+				benchmarkFn: () => {
+					writeDeepTree(tree, changedLeafValue);
+				},
+				after: () => {
+					const expected = generateDeepSimpleTree(numberOfNodes, changedLeafValue);
+					assert.deepEqual(tree, expected);
+				},
+			});
+		}
+
+		for (const [numberOfNodes, benchmarkType] of nodesCountWide) {
+			let tree: WideTreeNode;
+			benchmark({
+				type: benchmarkType,
+				title: `Remove and insert end value at leaf of ${numberOfNodes} Wide tree`,
+				before: () => {
+					// Setup
+					tree = generateWideSimpleTree(numberOfNodes, leafValue);
+				},
+				benchmarkFn: () => {
+					writeWideSimpleTreeNewValue(tree, changedLeafValue, tree.length - 1);
+				},
+				after: () => {
+					const actual = tree[tree.length - 1];
+					assert.equal(actual, changedLeafValue);
+				},
+			});
+		}
+
+		for (const [numberOfNodes, benchmarkType] of nodesCountWide) {
+			let tree: WideTreeNode;
+			benchmark({
+				type: benchmarkType,
+				title: `Remove and insert first value at leaf of ${numberOfNodes} Wide tree`,
+				before: () => {
+					// Setup
+					tree = generateWideSimpleTree(numberOfNodes, leafValue);
+				},
+				benchmarkFn: () => {
+					writeWideSimpleTreeNewValue(tree, changedLeafValue, 0);
+				},
+				after: () => {
+					const actual = tree[0];
+					assert.equal(actual, changedLeafValue);
+				},
+			});
+		}
+
+		for (const [numberOfNodes, benchmarkType] of nodesCountWide) {
+			let tree: WideTreeNode;
+			benchmark({
+				type: benchmarkType,
+				title: `Move second leaf to begining of ${numberOfNodes} Wide tree`,
+				before: () => {
+					// Setup
+					tree = generateWideSimpleTree(numberOfNodes, leafValue);
+					writeWideSimpleTreeNewValue(tree, changedLeafValue, 1);
+				},
+				benchmarkFn: () => {
+					tree.moveToIndex(0, 1);
+				},
+				after: () => {
+					// Even number of iterations cancel out, so this validation only works after odd numbers of iterations.
+					// Correctness mode always does a single iteration, so just validate that case.
+					if (!isInPerformanceTestingMode) assert.equal(tree[0], changedLeafValue);
+				},
+			});
+		}
+
+		for (const [numberOfNodes, benchmarkType] of nodesCountWide) {
+			let tree: WideTreeNode;
+			benchmark({
+				type: benchmarkType,
+				title: `Move next-to-last leaf to end of ${numberOfNodes} Wide tree`,
+				before: () => {
+					// Setup
+					tree = generateWideSimpleTree(numberOfNodes, leafValue);
+					writeWideSimpleTreeNewValue(tree, changedLeafValue, tree.length - 2);
+				},
+				benchmarkFn: () => {
+					tree.moveToIndex(tree.length - 2, tree.length - 1);
+				},
+				after: () => {
+					const actual = tree[tree.length - 1];
+					assert.equal(actual, changedLeafValue);
```

Comment:
the same fix you did to the above test is also needed here.

Reply:
On it